ws-client/src/transport.rs

```diff
@@ -146,43 +159,8 @@ pub enum WsConnectError {
 
 /// Creates a new WebSocket connection based on [`WsConfig`](crate::WsConfig) represented as a Sender and Receiver pair.
 pub async fn websocket_connection(config: WsConfig<'_>) -> Result<(Sender, Receiver), WsHandshakeError> {
```

Comment:
if you like the change i.e, provide `TryFrom<WsConfig> for WebSocketTransportClient { ... }` then I think we could remove this function.